tests/gnrc_udp/udp.c

```diff
@@ -97,14 +109,14 @@ static void send(char *addr_str, char *port_str, char *data_len_str, unsigned in
         return;
     }
     /* parse port */
-    port = atoi(port_str);
-    if (port == 0) {
+    port = valid_port(strtoul(port_str, &conversion_end, 0));
+    if (*conversion_end != '\0' && port != 0) {
         puts("Error: unable to parse destination port");
         return;
     }
 
-    data_len = atoi(data_len_str);
-    if (data_len == 0) {
+    data_len = strtoul(data_len_str, &conversion_end, 0);
+    if (*conversion_end != '\0') {
```

Comment:
Sleeping over it, let's just focus on fixing this one for now for the release. The other stuff should also be fixed in applications like [`examples/gnrc_networking`](https://github.com/RIOT-OS/RIOT/blob/7484af6d369a6af44671e1564ee8457b8f033aba/examples/gnrc_networking/udp.c)